library/pkparse.c

```diff
@@ -1144,6 +1163,28 @@ static int pk_parse_key_sec1_der(mbedtls_pk_context *pk,
 #endif /* MBEDTLS_USE_PSA_CRYPTO */
     }
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+    curve = mbedtls_ecc_group_to_psa(eck->grp.id, &curve_bits);
+    psa_set_key_type(&attributes, PSA_KEY_TYPE_ECC_KEY_PAIR(curve));
+    /* Setting largest masks for usage and key algorithms */
+    psa_set_key_usage_flags(&attributes, PSA_KEY_USAGE_SIGN_HASH |
+                            PSA_KEY_USAGE_SIGN_MESSAGE |
+                            PSA_KEY_USAGE_EXPORT);
+#if defined(MBEDTLS_ECDSA_DETERMINISTIC)
+    psa_set_key_algorithm(&attributes,
+                          PSA_ALG_DETERMINISTIC_ECDSA(PSA_ALG_ANY_HASH));
+#else
+    psa_set_key_algorithm(&attributes, PSA_ALG_ECDSA(PSA_ALG_ANY_HASH));
+#endif
```

Comment:
Note: this is the part that wouldn't work with RSA keys, as we'd want to allow encryption, PKCS#1 v1.5 signature, and PSS signature (aka PKCS#1 v2.1), and PSA only support setting up to two algorithms. But for ECC keys it's OK, as PK only supports doing ECDSA with them.